integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -590,7 +590,7 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
    * detects whether double or decimal column is part of dictionary_exclude
    */
   def isStringAndTimestampColDictionaryExclude(columnDataType: String): Boolean = {
-    val dataTypes = Array("string", "timestamp")
+    val dataTypes = Array("string", "timestamp", "date")
```

Comment:
@anubhav100 Please check my jira comment
this is wrong you should not allow adding timestamp and date in dictionary_exclude